src/App.jsx

```diff
@@ -13,46 +15,95 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORT_GOODS_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_GOODS_BY_LENGTH = 'Sort by length';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function updateGoods(goods, sortBy, isReversed) {
+  let prepearedGoods = [...goods];
+
+  if (sortBy) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SORT_GOODS_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+        case SORT_GOODS_BY_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isGoodsReversed, setIsGoodsReversed] = useState(false);
+  const sortedGoods
+    = updateGoods(goodsFromServer, sortBy, isGoodsReversed);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortBy !== SORT_GOODS_ALPHABETICALLY,
+          })}
+          onClick={() => setSortBy(SORT_GOODS_ALPHABETICALLY)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortBy !== SORT_GOODS_BY_LENGTH,
+          })}
+          onClick={() => setSortBy(SORT_GOODS_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isGoodsReversed,
+          })}
+          onClick={() => setIsGoodsReversed(!isGoodsReversed)}
+        >
+          Reverse
+        </button>
+
+        {(sortBy || isGoodsReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setIsGoodsReversed(false);
+              setSortBy('');
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)